lib/cloud_controller/deployment_updater/actions/finalize.rb

```diff
@@ -0,0 +1,55 @@
+require 'cloud_controller/deployment_updater/actions/cleanup_web_processes'
+module VCAP::CloudController
+  module DeploymentUpdater
+    module Actions
+      class Finalize
+        attr_reader :deployment, :deploying_web_process, :app
+
+        def initialize(deployment)
+          @deployment = deployment
+          @app = deployment.app
+          @deploying_web_process = deployment.deploying_web_process
+        end
+
+        def call
+          promote_deploying_web_process
+
+          CleanupWebProcesses.new(deployment, deploying_web_process).call
+
+          update_non_web_processes
+          restart_non_web_processes
+          deployment.update(
+            state: DeploymentModel::DEPLOYED_STATE,
+            status_value: DeploymentModel::FINALIZED_STATUS_VALUE,
+            status_reason: DeploymentModel::DEPLOYED_STATUS_REASON
+          )
+        end
+
+        private
+
+        def promote_deploying_web_process
+          deploying_web_process.update(type: ProcessTypes::WEB)
```

Comment:
Not sure if there is a point to this.  Tests do not fail if I comment it out.

Reply:
I think this is an artifact of 'webish' processes: https://github.com/cloudfoundry/cloud_controller_ng/commit/617fadec7210213afa087806196fa245287e97a0#diff-23c8e0c518bc3fdeaa2541c3275a7c737543c5efc35962d98494aa265c509446R74

I'm going to delete

Reply:
I see the name and behaviour were taken from the old implementation, interesting that the method name here really doesn't match the code.

The only thing I can think of is that would updating this trigger some Model hooks that would do something with diego thats important?

Reply:
Yep, that's how I understand it.